src/controllers/touristCenterController.js

```diff
@@ -0,0 +1,120 @@
+      const newTouristCenter = {
+        countryId, gallery, name: newName, location, about
+      };
+      const country = await db.findCountry(countryId);
+      if (!country) return res.status(404).json({ status: 404, error: "Country does not exist" });
+      const centerName = await db.findTouristCenter(newName);
+      if (centerName) return res.status(409).json({ status: 409, message: "This Tourist center already exists." });
+      const createdTouristCenter = await db.addTouristCenter(newTouristCenter);
+      return res.status(201).json({ status: 201, message: "A Tourist Center has been added.", data: createdTouristCenter, });
+    } catch (error) {
+      return res.status(500).json({ status: 500, error: "Server error." });
+    }
+  }
+
+  /**
+   * @param {object} req - The user request object
+   * @param {object} res - The user response object
+   * @returns {object} Success message
+   */
+  static async getAllTouristCenters(req, res) {
+    try {
+      const touristCenters = await db.listTouristCenters();
+      return res.status(200).json({ status: 200, message: "Successfully retrived all Tourist Centers", data: touristCenters, });
+    } catch (error) {
+      return res.status(500).json({ status: 500, error: "Server error." });
+    }
+  }
+
+  /**
+   * @param {object} req - The user request object
+   * @param {object} res - The user response object
+   * @returns {object} Success message
+   */
+  static async getTouristCenter(req, res) {
+    try {
+      const { id } = req.params;
+      const { error } = validateId({ id });
+      if (error) return res.status(400).json({ status: 400, error: error.message });
+      const touristCenter = await db.findTouristCenterById(id);
+      if (!touristCenter) return res.status(404).json({ status: 404, error: "Tourist Center not found" });
+      return res.status(200).json({ status: 200, message: "Successfully retrived Tourist Center.", data: touristCenter, });
+    } catch (error) {
+      return res.status(500).json({ status: 500, error: "Server error." });
+    }
+  }
+
+  /**
+   * @param {object} req - The user request object
+   * @param {object} res - The user response object
+   * @returns {object} Success message
+   */
+  static async updateTouristCenter(req, res) {
+    try {
+      const { id } = req.params;
+      const {
+        countryId, gallery, name, location, about
+      } = req.body;
+      const { error } = validateId({ id, countryId });
+      if (error) return res.status(400).json({ status: 400, error: error.message });
+      const oldCenter = await db.findTouristCenterById(id);
+      if (!oldCenter) return res.status(404).json({ status: 404, error: "Tourist Center not found" });
+      if (countryId) {
+        const country = await db.findCountry(countryId);
+        if (!country) return res.status(404).json({ status: 404, error: "Country does not exist" });
+      }
+      let newname;
+      if (name) {
```

Comment:
i dont think this body of validation code 
```
if (name) {
        newname = name[0].toUpperCase() + name.slice(1).toLowerCase();
        req.body.name = newname;
        const centerName = await db.findTouristCenter(newname);
        if (centerName) return res.status(409).json({ status: 409, message: "This Tourist center already exists." });
      }
```
is needed for an update function, it limits the full capabilities, remove it.

Reply:
Ok. I'll have to leave the following part tho just to make sure the name I'm updating to the database is capitalized.
```
if (name) {
        newname = name[0].toUpperCase() + name.slice(1).toLowerCase();
        req.body.name = newname;
}
```